src/ExportMap.js

```diff
@@ -346,7 +348,7 @@ ExportMap.parse = function (path, content, context) {
     docStyleParsers[style] = availableDocStyleParsers[style]
   })
 
-  const source = makeSourceCode(content, ast)
+  const source = context.id === 'no-deprecated' ? makeSourceCode(content, ast) : null
```

Comment:
Is there a way to make this metadata the rule can supply rather than hardcoding a single rule name?

Reply:
yeah, I'll take a look

Reply:
Passing an argument looks better, it seems rule `id` isn't passed in ESlint 2-3.

Reply:
but more changes are required... 😓

Reply:
maybe instead of skipping the comments, we could make them lazy?

ie, instead of using properties, use functions that lazily create the source code object as needed (but memoize it across calls)?

Reply:
I'm afraid it's too late, we don't have required data (content, ast) when we need it, otherwise we have to store them as well